Webpack Build Errors are Swallowed By Unlink Errors

**Describe the bug**

When you run a prod build, if there is a legitimate build error (e.g due to a typing error in your project) the error is hidden and you instead get an error that the map file could not be deleted.

**To Reproduce**

Steps to reproduce the behavior:

1. Setup a project and connect appsignal webpack (e.g. an Angular project)
2. Intentionally make a build error
3. Run a prod build

**Expected behavior**

That the ts error is shown instead of the map unlink error

**Screenshots**

If I run a build with appsignal webpack installed my build output looks like this:

![image](https://user-images.githubusercontent.com/9100419/104139206-babf6680-53f9-11eb-99db-5b60bb198dd8.png)

If I update `AppSignalPlugin.deleteFiles` in my `node_modules` to add a `catch` to the `unlink` as follows

![image](https://user-images.githubusercontent.com/9100419/104139227-f1957c80-53f9-11eb-98c9-8a8063cf9d0a.png)

I can see the actual error again as normal:

![image](https://user-images.githubusercontent.com/9100419/104139257-2b668300-53fa-11eb-89fa-2800a61d82ba.png)

I have made a PR which adds this catch block here:

https://github.com/appsignal/appsignal-javascript/pull/466

@zbarbuto Amazing! I used your example app to test your PR and I see how it goes wrong now. Thanks for the wait and the help getting this all working and reproducible. We're going to look at when to release this update along with #468. 

Here's the output from `yarn`. I used yarn to test this in our test setup as the webpack package uses yarn as well. It was easier to test that way for me:

```
⚠️ [AppsignalPlugin]: unable to delete 'main.js.map. File does not exist. it may not have been created due to a build error.
⚠️ [AppsignalPlugin]: unable to delete 'polyfills.js.map. File does not exist. it may not have been created due to a build error.
⚠️ [AppsignalPlugin]: unable to delete 'runtime.js.map. File does not exist. it may not have been created due to a build error.
⚠️ [AppsignalPlugin]: unable to delete 'styles.css.map. File does not exist. it may not have been created due to a build error.
⚠️ [AppsignalPlugin]: unable to delete 'vendor.js.map. File does not exist. it may not have been created due to a build error.

Error: src/app/app.component.ts:11:3 - error TS2322: Type 'string' is not assignable to type 'boolean'.

11   brokenValue: boolean = 'string';
     ~~~~~~~~~~~
```

--- a/packages/webpack/src/index.ts
+++ b/packages/webpack/src/index.ts
@@ -1,8 +1,8 @@
 import axios, { AxiosInstance } from "axios"
-import { Compilation, Compiler, Stats, WebpackPluginInstance } from "webpack"
 import FormData from "form-data"
 import fs from "fs/promises"
 import path from "path"
+import { Compilation, Compiler, Stats, WebpackPluginInstance } from "webpack"
 
 /**
  * This plugin borrows heavily from https://github.com/40thieves/webpack-sentry-plugin
@@ -164,15 +164,19 @@ export class AppsignalPlugin implements WebpackPluginInstance {
         )
 
         if (filePath) {
-          return fs.unlink(filePath)
+          return fs.unlink(filePath).catch(() => this.logFileRemoveError(name))
         } else {
-          console.warn(
-            `⚠️ [AppsignalPlugin]: unable to delete '${name}. File does not exist. it may not have been created due to a build error.`
-          )
+          this.logFileRemoveError(name)
         }
       })
       .filter(el => el)
 
     return Promise.all(promises)
+  }
+
+  logFileRemoveError(name: string) {
+    console.warn(
+      `⚠️ [AppsignalPlugin]: unable to delete '${name}. File does not exist. it may not have been created due to a build error.`
+    )
   }
 }

